### DISCUSSION TITLE
Expanding shared ports protocols to TCP

### DISCUSSION THREAD

**Discussion by Conor-Behard333** [coder/coder#12964] [Feature Request]
Hi,

I have coderdeployed on a k8s cluster and a workspace using DIND. I have deployed an SQL in the workspace using docker-compose, exposing 1433 and 1443 (TCP). Our team wants to be able to access it from outside of the workspace. Port sharing seem like it would be the solution to this, however, the current port sharing feature only supports HTTP(S). 

Are there any plans to allow other protocols (TCP, UDP) to be used for this feature?

**Comment by ammario** [comment]
It's a worthwhile feature request with an obvious user experience in our existing `port-forward` command:

```
coder port-forward <other user>/<workspace>.<agent> --tcp 1433:1433
```

Right now, `coder port-forward` is used by:

* Users accessing their own ports
* Admins accessing other user ports

It seems natural to extend that to

* Users accessing other user ports (when allowed)

**Comment by matifali** [comment]
@spikecurtis @ethanndickson can we support tcp using Coder Desktop?

**Comment by spikecurtis** [comment]
Coder Desktop VPN supports any protocol that runs over IPv6 including TCP, but we don’t support sharing. You can only access your own workspaces at present.
